client/src/lib/ShortcodeSerialiser.js

```diff
@@ -0,0 +1,134 @@
+import i18n from 'i18n';
+
+/**
+ * The following is Copyright (c) 2012, Ernesto Méndez (der@der-design.com)
+ * Licensed under the MIT License
+ * https://github.com/mendezcode/shortcode-parser/blob/master/lib/shortcode-parser.js
+ * Differences in SS shortcodes:
+ *  - SS shortcodes do not use `/]` to close shortcodes
+ *  - Properties may be `,` separated instead of space separated, in order to
+ *  prevent urlencoding issues
+ *  - Shortcodes cannot have spaces between opening brace and name. i.e. `[name`
+ */
+// Used to match outer regexp and get attrs as a string
+// All attrs extracted into matches[1]
+// eslint-disable-next-line max-len
+const SHORTCODE_ATTRS = /((?:[,\s]+(?:[a-z0-9\-_]+)=(?:(?:[a-z0-9\-_]+)|(?:\d+\.\d+)|(?:'[^']*')|(?:"[^"]*")))*)/
+  .toString()
+  .slice(1, -1);
+// Used to extract individual items from above regexp
+// Each item matches[1] is key, and matches[2] || matches[3] || matches[4] || matches[5] is value
+// eslint-disable-next-line max-len
+const SHORTCODE_ATTR = /[,\s]+([a-z0-9\-_]+)=(?:([a-z0-9\-_]+)|(\d+\.\d+)|(?:'([^']*)')|(?:"([^"]*)"))/;
```

Comment:
SHORTCODE_ATTR and SHORTCODE_ATTRS ate away at my life. Never again javascript regexps!